distributed/worker.py

```diff
@@ -161,8 +160,8 @@ def heartbeat(self):
             self.heartbeat_active = True
             logger.debug("Heartbeat: %s" % self.address)
             try:
-                if proc:
-                    memory_info = proc.memory_info()
+                if psutil:
+                    memory_info = psutil.Process().memory_info()
```

Comment:
I think instantiating a new Process object each time this method is called is a bit wasteful. Doing it in _start() would be sufficient.